pkg/internal/sqlprune/sqlparser_test.go

```diff
@@ -1,56 +1,142 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Grafana Labs
+ * SPDX-License-Identifier: Apache-2.0
+ */
 package sqlprune
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/xwb1989/sqlparser"
 )
 
-var tableRequired = map[string]bool{
-	"DELETE": true,
-	"INSERT": true,
-	"UPDATE": true,
-}
+func TestSQLExtraction(t *testing.T) {
+	type result struct {
+		op    string
+		table string
+	}
 
-func queryIsNotParsable(query string) bool {
-	_, err := sqlparser.Parse(query)
-	return err != nil
-}
+	t.Run("test SELECT", func(t *testing.T) {
+		tests := map[string]result{
+			"SELECT t.id, t.name FROM ACCESS_TOKENS t, SECURITY_POLICIES sp WHERE sp.id=t.security_policy_id AND sp.org_id=?": {op: "SELECT", table: "ACCESS_TOKENS,SECURITY_POLICIES"},
+			"SELECT * FROM TABLE WHERE FIELD=1234": {op: "SELECT", table: ""},
+			"SELECT * FROM ZOOM WHERE FIELD=1234":  {op: "SELECT", table: "ZOOM"},
+			`SELECT
```

Comment:
I tried to add some realistic select statements, with joins and nested selects. I may look for more convoluted scenarios.